commit c64a4b54dd16537ad0c609037e96d295a60d9ace
Author: Florian Festi <ffesti@redhat.com>
Date:   Mon Jun 30 13:30:38 2014 +0200

    Fix handling of hardlinks during rpmbuild
    
    After looping over the hardlinks and writing their headers entries to the
    archive we need to return to the first entry to make sure we do not leave out
    other groups of hardlinked files that start between the group we are currently
    processing
    
    (cherry picked from commit b7b8b3343f8668f77baf8f2cddbdd99d9d62c1f4)

diff --git a/lib/rpmfi.c b/lib/rpmfi.c
index 27186dd..ed51d44 100644
--- a/lib/rpmfi.c
+++ b/lib/rpmfi.c
@@ -1819,6 +1819,7 @@ static int iterWriteArchiveNextFile(rpmfi fi)
 		return rc;
 	    }
 	}
+	rpmfiSetFX(fi, hardlinks[0]);
     } else {
 	int rc = rpmfiArchiveWriteHeader(fi);
 	if (rc) {
